app/controllers/pushes_controller.rb

```diff
@@ -0,0 +1,359 @@
+    end
+  end
+
+  def index
+    unless Settings.enable_logins
+      redirect_to :root
+      return
+    end
+
+    @filter = params[:filter]
+    allowed_kinds = ["text"]
+    if Settings.enable_file_pushes
+      allowed_kinds << "file"
+    end
+
+    if Settings.enable_url_pushes
+      allowed_kinds << "url"
+    end
+
+    @pushes = if @filter
+      Push.includes(:audit_logs)
+        .where(user_id: current_user.id, expired: @filter == "expired", kind: allowed_kinds)
+        .page(params[:page])
+        .order(created_at: :desc)
+    else
+      Push.includes(:audit_logs)
+        .where(user_id: current_user.id)
+        .page(params[:page])
+        .order(created_at: :desc)
+    end
+  end
+
+  private
+
+  def set_push
+    @push = Push.includes(:audit_logs).find_by!(url_token: params[:id])
+  rescue ActiveRecord::RecordNotFound
+    # Showing a 404 reveals that this Secret URL never existed
+    # which is an information leak (not a secret anymore)
+    # We also don't want data in general. We entirely delete old pushes that:
+    # 1. have expired (payloads already deleted long ago)
+    # 2. are anonymous/not linked to a user account (audit log not needed)
+    # Old, expired & anonymous pushes have no value to anybody.
+    # When not found, show the 'expired' page so even very old secret URLs
+    # when clicked they will be accurate - this secret URL has expired.
+    # No easy fix for JSON unfortunately as we don't have a record to show.
+    respond_to do |format|
+      format.html { render template: "pushes/show_expired", layout: "naked" }
+    end
+  end
+
+  def push_params
+    case params.dig(:push, :kind)
+    when "url"
+      params.require(:push).permit(:kind, :name, :expire_after_days, :expire_after_views,
+        :retrieval_step, :payload, :note, :passphrase)
+    when "file"
+      params.require(:push).permit(:kind, :name, :expire_after_days, :expire_after_views, :deletable_by_viewer,
+        :retrieval_step, :payload, :note, :passphrase, files: [])
+    else
+      params.require(:push).permit(:kind, :name, :expire_after_days, :expire_after_views, :deletable_by_viewer,
+        :retrieval_step, :payload, :note, :passphrase)
+    end
+  rescue => e
+    Rails.logger.error("Error in push_params: #{e.message}")
+    raise e
+  end
+
+  def print_preview_params
+    params.permit(:id, :locale, :message, :show_expiration, :show_id)
+  end
+
+  def set_kind_by_tab
+    # Track which tab to show
+    if params.key?("tab")
+      if params["tab"] == "text"
+        @push.kind = "text"
+        @text_tab = true
+      elsif params["tab"] == "files"
+        @push.kind = "file"
+        @files_tab = true
+      elsif params["tab"] == "url"
+        @push.kind = "url"
+        @url_tab = true
+      else
+        @push.kind = "text"
+        @text_tab = true
+      end
+    else
+      @push.kind = "text"
+      @text_tab = true
+    end
+  end
+
+  def check_allowed
```

Comment:
Previously, `Settings.enable_logins && Settings.enable_file_pushes` checks were done when routing different requests to different controllers. Now, we have only one controller for HTML requests, so we need to handle this logic in this controller. 

Also, authorization was easier to define by using `before_action` callback, but now different actions has different rules for different Push kinds. 

So, I have put all logic into one method. Also, I don't want to split it into different methods not to miss any rule set before.

Reply:
I like this centralized.  I also realize we need to clean up all these settings in the future.

Should we also consider the user setting for [Settings.allow_anonymous](https://github.com/pglombardo/PasswordPusher/blob/master/config/settings.yml#L113-L125) here?